app/src/main/java/com/portfolio/tasky/usecases/NetworkChangeReceiver.kt

```diff
@@ -0,0 +1,62 @@
+package com.portfolio.tasky.usecases
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.net.ConnectivityManager
+import android.net.ConnectivityManager.NetworkCallback
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+
+
+
+
+
+class NetworkChangeReceiver constructor(internetStatus: InternetStatus) : BroadcastReceiver() {
```

Comment:
Remember, if a usecase uses android-specific dependencies, then is should be an interface, where the implementation takes care of the android part. But the interface doesn't have Android logic

Check: https://www.youtube.com/watch?v=TzV0oCRDNfM

Reply:
Here I have used NetworkChangeReceiver as my impl class, I used InternetStatus interface as abstraction. When a network event generates in NetworkChangeReceiver my EntryActivity class receives it through abstraction/interface's overridden methods, internetAvailable(), internetNotAvailable() ..I think for testing I can use some FakeInternetStatus interface to test it out..

I am not sure what I am missing 😕

Reply:
Now I get it. Earlier I had used interface but couldn't get it working, with new implementation I have used LiveData which has a type of an interface called NetworkStatus. I hope this is allowed..